packages/auth/demo/src/index.js

```diff
@@ -493,6 +494,106 @@ function onInitializeRecaptchaConfig() {
   initializeRecaptchaConfig(auth);
 }
 
+/**
+ * Updates the displayed validation status for the inputted password.
+ */
+function onValidatePassword() {
+  /**
+   * Updates the displayed status for a requirement.
+   * @param {string} id The ID of the DOM element displaying the requirement status.
+   * @param {boolean | undefined} status Whether the requirement is met.
+   */
+  function setRequirementStatus(id, status) {
+    // Hide the requirement if the status does not include it.
+    if (status === undefined) {
+      $(id).hide();
+      return;
+    }
+
+    if (status) {
+      $(id).removeClass('list-group-item-danger');
+      $(id).addClass('list-group-item-success');
+    } else {
+      $(id).removeClass('list-group-item-success');
+      $(id).addClass('list-group-item-danger');
+    }
+    $(id).show();
+  }
+
+  const password = $('#password-validation-password').val();
+  validatePassword(auth, password).then(
+    status => {
+      const passwordPolicy = status.passwordPolicy;
+      const customStrengthOptions = passwordPolicy.customStrengthOptions;
+
+      // Only show options required by the password policy.
+      if (customStrengthOptions.minPasswordLength) {
+        $('#password-validation-min-length').text(
+          customStrengthOptions.minPasswordLength
+        );
+      }
+      if (customStrengthOptions.maxPasswordLength) {
+        $('#password-validation-max-length').text(
+          customStrengthOptions.maxPasswordLength
+        );
+      }
+      if (customStrengthOptions.containsNonAlphanumericCharacter) {
+        $('#password-validation-allowed-non-alphanumeric-characters').attr(
+          'data-original-title',
+          passwordPolicy.allowedNonAlphanumericCharacters
+        );
+      }
+      setRequirementStatus(
+        '#password-validation-meets-min-length',
+        status.meetsMinPasswordLength
+      );
+      setRequirementStatus(
+        '#password-validation-meets-max-length',
+        status.meetsMaxPasswordLength
+      );
+      setRequirementStatus(
+        '#password-validation-contains-lowercase',
+        status.containsLowercaseLetter
+      );
+      setRequirementStatus(
+        '#password-validation-contains-uppercase',
+        status.containsUppercaseLetter
+      );
+      setRequirementStatus(
+        '#password-validation-contains-numeric',
+        status.containsNumericCharacter
+      );
+      setRequirementStatus(
+        '#password-validation-contains-non-alphanumeric',
+        status.containsNonAlphanumericCharacter
+      );
```

Comment:
can you use a for loop here to iterate thru the keys using `Object.keys(passwordPolicy)`.  Using the property as ID is acceptable to me, but you can also manipulate the string if you want with a password-validation prefix.

Reply:
Is it okay that the IDs would then be in camelCase instead of kebab-case like the other IDs?

Reply:
you can try use a regex replace/a helper method to format them to kabob like strings.

Reply:
Done.